dev/package.sh

```diff
@@ -8,6 +8,11 @@ LINUX_OS=$(. /etc/os-release && echo ${ID})
 VERSION=$(. /etc/os-release && echo ${VERSION_ID})
 ARCH=`uname -m`
 
+if [[ -n "$JAVA_HOME" ]]; then
+  echo "JAVA_HOME needs to be set for Gluten Compilation"
+  exit 1
+fi
```

Comment:
s/Compilation/compilation

What's the major consideration to make JAVA_HOME required?

Reply:
General best practice and mainly to avoid inconsistency since if the system has JAVA_HOME set gluten compilation will happen on that but third party libs will be generated with the Java on PATH of the system.

Reply:
@CodenameGHOST007 

I am worried whether this change break user's build processes much. Can we separate this change out to future PRs? Thanks.

Reply:
Removed these changes. Will create a separate issue to track it. Ideally we should find and set JAVA_HOME itself before starting the build like spark and all do.